### DISCUSSION TITLE
Feature suggestions/feedback

### DISCUSSION THREAD

**Discussion by Therightroad2009** [SoftbearStudios/mk48#192] [General]
Hello everyone, i came out with those ideas and i'd like to suggest them
### ****Periscope Depth button (submarines)****
This would be a button just like Active sensors, etc. Pressing this would bring you to the Periscope depth of the submarine **(e.g. Akula 19 meters)** and would automatically turn on an Visual sensor **(which cant be used underwater in subs)** but going to this depth will make you
so much easier to detect with Visual sensors.
Basically the sensor will allow you to detect ships with an visual sensor (**larger detection range because periscopes have zoom**)
an automatically know what the ship is.
### ****Map Feature****
There would be the feature of an map, where you can open and close it, and even adjust the size of the map screen.
In the map you would see the location of your teammates **(only works in fleets)** and your own position. Thats all i have to say.
### "Ultra Quiet" button for submarines
I heard about this on another discussion, and here i'll give my opinion on this: 
I really recommend it, because it's hard to ajust the exact speed where the cavitation stops which is kind of annoying. And this button would make all of it in 1 second.

Now im going to say some stuff that i don't agree with in the game.
1. When i have active sensors off as a submarine, i can't detect oil rigs (unless they're close) and this does really annoy me, when i shooted all of my weapons to a lvl-7-9 ship, everything i shooted dissapeared because of an oil rig (the ship was really just so close and the oil rig in the middle, and i didn't even detect the oil rig, so that's my point, why did i detect the ship and not the oil rig?
MY OPINION: Oil rigs should be detected easier or being at more far distance (high frequency sonars also exist) in real life it would be detected.

2. planes/helis detecting subs too easily, Imagine being quiet in a sub, then you get shooted from nowhere being even at silent speeds and sensors off, well it is the planes/helis, heres my pov on how i see this working, It doesn't matter your speed or whatever if you are a sub, if a heli or plane just does overfly you, they detect you, it's like helis and planes know everything else without having to detect the subs. 
And this is very annoying. I'd like to know if this is being fixed.

Thats all of it, greets.

**Comment by finnbear** [comment]
Thanks for the comprehensive feedback! :+1: 

> When i have active sensors off as a submarine, i can't detect oil rigs (unless they're close) and this does really annoy me

This is sort of intentional. When you disable active sensors, you are disabling active sonar. Active sonar allows you to see objects in the water by bouncing sound waves off of them. With it off, all you have is passive sonar, which operates by detecting noises from objects. Ships make noise (engines, propellers, machinery, waves), so are more visible on passive sonar. Oil rigs don't really make much noise, and you are in a submerged metal tube with no windows.

In multiple documented cases, submarines have collided with stationary objects, usually undersea moutains.
![image](https://user-images.githubusercontent.com/20015102/174695468-814b38ee-6c23-49f1-94d3-30a66ac4317b.png)

>  MY OPINION: Oil rigs should be detected easier or being at more far distance (high frequency sonars also exist) in real life it would be detected.

In Mk48.io, active sonar is all-or-nothing. We didn't want to complicate the game by adding multiple types, even if it would mean you could use close-range active sonar without being detected as much as long-range active sonar.

What we can do is to slightly increase the amount of noise that oil rigs make so they become more visible on passive sonar.

> planes/helis detecting subs too easily

You're absolutely right! We plan to make it more difficult for fast-moving (relatively-non-stationary) helicopters to detect submarines, as they must use dipping sonar in real life. As for planes, we might limit detection of subs that are not actively reloading (we do need to maintain *some* ability for battleships to defend themselves against subs). It's worth mentioning that some IRL aircraft are equipped with magnetic anomaly detectors and copious amounts of sonobouys, albeit not the in game planes.

> Map Feature

Ability to know where your fleet members are sounds good. Although a map might involve sending too much data (all the terrain, and multiple players), we might make it so you can click a fleetmate's name to get an in-game arrow pointing to them. We'll also make it easier for you to communicate your coordinates in team chat (with a click or two, instead of typing them)

> Periscope Depth button (submarines)

This seems a bit problematic because, in the vast majority of cases (all cases except the Type VIIC deck gun, which can only be fired while surfaced), periscope depth would be just as good as surfaced for the sub, but would make it immune from various airborne weapons. As a result, this change would be a significant and not-yet-justified re-balancing in favor of subs.

**Reply by Therightroad2009** [reply]
Thanks for the answer, i've wanted an opinion on this, i agree with helis/planes using MAD/ dipping sonar sensors
what i mean about the oil rigs detection is to make them easier to detect as you said and i'd like that close range active sonar (it's just like mine avoidance systems in subs irl) a bug i also want to mention is that when Helis/ planes shoot torpedos to subs being close the torpedo never gets to hit the sub unless the torpedo is fired from larger distance, i don't know if this is intentional because of the torpedo falling from the air to the water but i'd like to know why is this happening if it is intentional or not.

**Reply by finnbear** [reply]
> i also want to mention is that when Helis/ planes shoot torpedos to subs being close the torpedo never gets to hit the sub unless the torpedo is fired from larger distance, i don't know if this is intentional because of the torpedo falling from the air to the water but i'd like to know why is this happening if it is intentional or not.

Yes, what you observe has to do with the torpedo starting at some altitude, hitting the water, and then descending to some depth (all of which takes time). This is fallible, as if the torpedo is dropped too late, it may cruise right over deep subs. While not precisely intentional, we are not necessarily going to change/fix this behavior, because the altitude system (as a result of trying to fit current altitude and desired altitude into one byte) hard to work with, and we don't necessarily want to rebalance in favor of planes.
